File: JavaScript/프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    let R=0, C=0,J=0,A=0
    
    survey.forEach((v,i)=>{
        const target = v[0]
        
        choices[i] = -(choices[i]-4)
        
        if(target=="R"||target=="T"){
            if(target=="R"){
                 R=R+choices[i]
            }
            else{
                R=R-choices[i]
            }
            return ;
        }
        
        if(target=="C" || target == "F"){
            if(target=="C"){
                C=C+choices[i]
            }
            else{
                C=C-choices[i]
            }
            
            return ;
        }
        
        if(target=="J" || target == "M"){
            if(target=="J"){
                J=J+choices[i]
            }
            else{
                J=J-choices[i]
            }
            
            return ;
        }
        
        if(target=="A" || target == "N"){
            if(target=="A"){
                A=A+choices[i]
            }
            else{
                A=A-choices[i]
            }
            
            return ;
        }
    })
    
    return `${R>=0?"R":"T"}${C>=0?"C":"F"}${J>=0?"J":"M"}${A>=0?"A":"N"}`
}

File: Kotlin/프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.kt
class Solution {
    fun solution(survey: Array<String>, choices: IntArray): String {
        var R = 0
            var C = 0
            var J = 0
            var A = 0

            survey.forEachIndexed { index, item ->
                val target: String = item[0].toString()
                choices[index] = -(choices[index] - 4)

                when(target) {
                    "R", "T" -> if(target == "R") R += choices[index] else R-= choices[index]
                    "C", "F" -> if(target == "C") C += choices[index] else C-= choices[index]
                    "J", "M" -> if(target == "J") J += choices[index] else J-= choices[index]
                    "A", "N" -> if(target == "A") A += choices[index] else A -= choices[index]
                }
            }

            var answer : String = ""
            answer += if (R>=0) "R" else "T"
            answer += if (C>=0) "C" else "F"
            answer += if (J>=0) "J" else "M"
            answer += if (A>=0) "A" else "N"

            return answer
    }
}